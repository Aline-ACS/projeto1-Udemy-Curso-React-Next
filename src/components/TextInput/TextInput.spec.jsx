import userEvent from '@testing-library/user-event';
import { TextInput } from './index';
const { render, screen } = require('@testing-library/react');

describe('<TextInput/>', () => {
    it('should have a value on searchValue', () => {
        const fn = jest.fn();

        render(<TextInput handleChange={fn} searchValue={'testando'}/>);

        const input = screen.getByPlaceholderText(/type your search/i);
        
        expect(input).toBeInTheDocument();
        expect(input.value).toBe('testando');
    });
    
    it('should call handleChange function on each key pressed', () => {
        const fn = jest.fn();

        render(<TextInput handleChange={fn} searchValue="um valor qualquer"/>);

        const valueInput = 'o valor';
        const input = screen.getByPlaceholderText(/type your search/i);


        userEvent.type(input, valueInput);
        
        expect(input.value).toBe('um valor qualquer');
        expect(fn).toHaveBeenCalledTimes(valueInput.length);
    });

    it('sholud match snapshot', () => {
        const fn = jest.fn();

        const {container} = render(<TextInput handleChange={fn} searchValue={'testando'}/>);

        expect(container.firstChild).toMatchSnapshot();
    });
});