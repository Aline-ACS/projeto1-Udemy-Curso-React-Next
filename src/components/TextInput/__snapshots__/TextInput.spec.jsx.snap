// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> sholud match snapshot 1`] = `
<input
  class="textinput"
  placeholder="Type your search"
  type="search"
  value="testando"
/>
`;
